### PR TITLE
Add support for `std::invoke`-like calls in `apply`.

### PR DESCRIPTION
**Pull request by jehelset**
Also adds noexcept spec via `is_nothrow_invocable` type-trait.

### CODE REVIEW

**Comment by jehelset** [comment]
@jfalcou i will add some more tests (hence draft), just wondering if general shape is ok.

**Comment by jfalcou** [comment]
OOh forgot to reply here.

The failures :

```
/Users/runner/work/kumi/kumi/include/kumi/utils/apply.hpp:16:41: error: no template named 'unwrap_reference_t' in namespace 'std'; did you mean 'unwrap_reference'?
      !std::is_same_v<std::decay_t<std::unwrap_reference_t<T &&>>, std::unwrap_ref_decay_t<T &&>>;
                                   ~~~~~^~~~~~~~~~~~~~~~~~
                                        unwrap_reference
/Users/runner/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/include/c++/v1/utility:644:8: note: 'unwrap_reference' declared here
struct unwrap_reference : __unwrap_reference<_Tp> { };
       ^
In file included from /Users/runner/work/kumi/kumi/test/doc/adapt.cpp:6:
In file included from /Users/runner/work/kumi/kumi/include/kumi/tuple.hpp:[15](https://github.com/jfalcou/kumi/actions/runs/7024999372/job/19114937475#step:4:16):
In file included from /Users/runner/work/kumi/kumi/include/kumi/utils.hpp:34:
/Users/runner/work/kumi/kumi/include/kumi/utils/apply.hpp:[16](https://github.com/jfalcou/kumi/actions/runs/7024999372/job/19114937475#step:4:17):73: error: no template named 'unwrap_ref_decay_t' in namespace 'std'; did you mean 'unwrap_ref_decay'?
      !std::is_same_v<std::decay_t<std::unwrap_reference_t<T &&>>, std::unwrap_ref_decay_t<T &&>>;
                                                                   ~~~~~^~~~~~~~~~~~~~~~~~
                                                                        unwrap_ref_decay
```

comes from the incomplete android STL, just use `typename unwrap<...>::type`

File: standalone/kumi/tuple.hpp [hunk 1]
```diff
@@ -528,7 +528,8 @@ namespace kumi
   namespace _{
     template<typename T>
     inline constexpr bool is_reference_wrapper_v =
-      !std::is_same_v<std::decay_t<std::unwrap_reference_t<T &&>>, std::unwrap_ref_decay_t<T &&>>;
+      !std::is_same_v<std::decay_t<typename std::unwrap_reference<T &&>::type>,
+                      typename std::unwrap_ref_decay<T &&>::typev>;
```
Comment (jfalcou) [inline standalone/kumi/tuple.hpp:532]: Typo on typev

File: include/kumi/utils/apply.hpp [hunk 2]
```diff
@@ -13,7 +13,8 @@ namespace kumi
   namespace _{
     template<typename T>
     inline constexpr bool is_reference_wrapper_v =
-      !std::is_same_v<std::decay_t<std::unwrap_reference_t<T &&>>, std::unwrap_ref_decay_t<T &&>>;
+      !std::is_same_v<std::decay_t<typename std::unwrap_reference<T &&>::type>,
+                      typename std::unwrap_ref_decay<T &&>::typev>;
```
Comment (jfalcou) [inline include/kumi/utils/apply.hpp:17]: Typo on typev

**Pull request merged**